pkg/istio-agent/xds-proxy.go

```diff
@@ -392,6 +465,29 @@ func getTLSDialOption(agent *Agent) (grpc.DialOption, error) {
 	return grpc.WithTransportCredentials(transportCreds), nil
 }
 
+func (p *XdsProxy) getRootCertificate(agent *Agent) (*x509.CertPool, error) {
+	var certPool *x509.CertPool
+	var err error
+	var rootCert []byte
+	var xdsCACertPath string
+	if agent.secOpts.FileMountedCerts {
+		xdsCACertPath = agent.proxyConfig.ProxyMetadata[MetadataClientRootCert]
+	} else {
+		xdsCACertPath = agent.FindRootCAForXDS()
```

Comment:
why doesn't FindRootCAForXDS encode the `MetadataClientRootCert` logic? The whole point of the FindRootCAForXDS is to avoid this type of logic proliferation

Reply:
Sure. It can. I will make that change.